pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -297,7 +335,13 @@ func (c *Cluster) makeDeployment(nodeName string, selection rookalpha.Selection,
 					},
 				},
 				Spec: v1.PodSpec{
-					NodeSelector:       map[string]string{v1.LabelHostname: nodeName},
+					Affinity: &v1.Affinity{
+						NodeAffinity:    osdProps.placement.NodeAffinity,
+						PodAffinity:     osdProps.placement.PodAffinity,
+						PodAntiAffinity: osdProps.placement.PodAntiAffinity,
+					},
+					Tolerations:        osdProps.placement.Tolerations,
+					NodeSelector:       map[string]string{v1.LabelHostname: osdProps.crushHostname},
```

Comment:
I see the node selector is cleared below if it's pvc based. For clarify, let's not set it here, then only set it later if !pvcBacked.